network/p2p/unicast/bandwidth_rate_limiter.go

```diff
@@ -0,0 +1,80 @@
+package unicast
+
+import (
+	"golang.org/x/time/rate"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/onflow/flow-go/network/message"
+)
+
+// BandWidthRateLimiterImpl unicast rate limiter that limits the bandwidth that can be sent
+// by a peer per some configured interval.
+type BandWidthRateLimiterImpl struct {
+	rateLimitedPeers *rateLimitedPeersMap
+	limiters         *rateLimiterMap
+	limit            rate.Limit
+	burst            int
+	now              GetTimeNow
+}
+
+// NewBandWidthRateLimiter returns a new BandWidthRateLimiterImpl. The cleanup loop will be started in a
+// separate goroutine and should be stopped by calling Close.
+func NewBandWidthRateLimiter(limit rate.Limit, burst int, now GetTimeNow) *BandWidthRateLimiterImpl {
+	l := &BandWidthRateLimiterImpl{
+		rateLimitedPeers: newRateLimitedPeersMap(rateLimiterTTL, cleanUpTickDuration),
```

Comment:
Using `rateLimiterTTL` here means `IsRateLimited` will report a peer as rate limited for `rateLimiterTTL` after it initially reached the limit or until `Allow` is called. Is that lockout intentional?

Reply:
`rateLimiterTTL` will guarantee us that we don't have stale keys in our map. A peer will only be "un-rate limited" if they have been inactive (TTL reached) or if Allow returns true.

Reply:
this problem still remains: if peer gets ratelimited `IsRateLimited` will short-circuit the `MessageAllowed`/`BandwidthAllowed` checks.